src/nrtk_explorer/library/dataset.py

```diff
@@ -1,24 +1,54 @@
-import kwcoco
-from pathlib import Path
-
-
-def load_dataset(path: str):
-    return kwcoco.CocoDataset(path)
-
+"""
+Module to load the dataset and get the image file path given an image id.
 
-dataset: kwcoco.CocoDataset = kwcoco.CocoDataset()
-dataset_path: str = ""
-
-
-def get_dataset(path: str, force_reload=False):
-    global dataset, dataset_path
-    if dataset_path != path or force_reload:
-        dataset_path = path
-        dataset = load_dataset(dataset_path)
-    return dataset
+Example:
+    dataset = get_dataset("path/to/dataset.json")
+    image_fpath = dataset.get_image_fpath("image_id")
+"""
 
+from functools import lru_cache
+from pathlib import Path
 
-def get_image_path(id: str):
-    dataset_dir = Path(dataset_path).parent
-    file_name = dataset.imgs[int(id)]["file_name"]
-    return str(dataset_dir / file_name)
+import json
+
+
+class DefaultDataset:
+    """Default dataset class to load the dataset and get the image file path given an image id."""
+
+    def __init__(self, path: str):
+        with open(path) as f:
+            self.data = json.load(f)
+            self.fpath = path
+            self.cats = {cat["id"]: cat for cat in self.data["categories"]}
+            self.anns = {ann["id"]: ann for ann in self.data["annotations"]}
+            self.imgs = {img["id"]: img for img in self.data["images"]}
+
+    def get_image_fpath(self, id_: str):
+        """Get the image file path given an image id."""
+        dataset_dir = Path(self.fpath).parent
+        file_name = self.imgs[int(id_)]["file_name"]
+        return str(dataset_dir / file_name)
+
+
+@lru_cache
```

Comment:
Consiter `@lru_cache(maxsize=1)` or just maintain module level `dataset` variable like before.

Reply:
What could be the drawbacks of caching more than one dataset? We do have the reload flag for re-reading.

Reply:
No biggie, but we could drop the `force_reload` arg and  `__load_dataset.cache_clear()` code if cache max was 1.  Effectivly we only cache 1 dataset with code as is.

Reply:
If we do that how can we reload the current dataset? The maybe cases where the user edits its dataset and wants to reload it with the same path. That was the reasoning of force_reload.

Reply:
Yes, I remember =)  Its a good idea but there is not way to actually trigger a reload in the GUI right now so I was trying to kill that code again.  =)

No biggie, I'm good with keeping the reload code because it will be nice to have sometime.

Reply:
I can see that core constructor sets the callback for current_dataset first, it then should invoke first `get_dataset(new_path, force=True)` and later other components they do this in their corresponding callback:

```
        if self.context.dataset is None:
            self.context.dataset = get_dataset(self.state.current_dataset, force_reload=True)
```